### ISSUE TITLE
What to do about Jekyll assets, with triple dashes

### ISSUE THREAD

**Issue by srawlins** [brigade/scss-lint#689]
Jekyll [specifies](http://jekyllrb.com/docs/assets/) that asset files, such as a .scss file, should start with two lines of triple dashes:

```
---
---
```

for some reason or other. Of course, these two lines make this an invalid SCSS file, which means the Sass gem can't parse it, which means we can't lint it. Anything to be done?

**Comment by sds** [comment]
This is definitely one of my biggest gripes with Jekyll. It changes the format of the file without changing the extension, but that's not going to change anytime soon, sadly.

The `haml-lint` project supports a `skip_frontmatter` option to work around this exact issue, but it bugs me that it's so specific to Jekyll. In a perfect world we would have a flag allowing you to specify a program you can preprocess a source file with before `scss-lint` lints it. This way you can just have `sed s/<some-regex>//` quickly cut out the frontmatter. 

If we're going to go down this route, I would prefer the more generic solution. We could call the option `preprocess_command` and have it take an array of arguments. Its input would be provided on STDIN and output should be sent to STDOUT. It would then be trivial for `scss-lint` to call it (I'm not worried about batching together files for performance in this case since it is already an exceptional situation).

Happy to accept a pull request along those lines.

**Comment by srawlins** [comment]
I like this idea. However, to make it a little more performant, I think we could couple it with a subdirectory listing. So something like:

```yaml
preprocess_command: strip-meta-stuff
preprocess_files:
    'some/folder/**'
    'another/directory/**/*.scss'
```

(Using the same glob syntax as `scss_files` and `exclude`.) This helps performance so that only a select few files are processed. The only shortcoming is you cant specify a preprocessor for directory A and another preprocessor for directory B. If a super user needs this, they just get to write a routing preprocessor.

I'll send a PR.

**Comment by sds** [comment]
One thing quickly worth mentioning is that stripping content will potentially offset line numbers for the document unless you are careful. In the case of Jekyll frontmatter, you need to replace each line with a blank line so line numbers don't shift. However, this in turn could trigger other lints (for example if we had a linter that checked for unnecessary leading whitespace).

**Comment by sds** [comment]
Implemented in #736.

**Comment by srawlins** [comment]
Thanks so much for releasing this in [0.47.0](https://github.com/brigade/scss-lint/tree/v0.47.0) so fast!
